README.md

```diff
@@ -1,2 +1,83 @@
+
+You'll connect using a string like: `ssh -A ec2-user@ec2-35-156-235-11.eu-central-1.compute.amazonaws.com`.
+
+The "-A" is NOT optional. It is required in order for the master node to be able to ssh
+into the worker nodes and for the nodes to be able to checkout private repos. In
+addition, your ssh-agent should have the key loaded. That means you should not pass the
+`-i` flag into ssh. Instead, run `ssh-add [your pem file]` before
+running ssh.
+
+This is an unfortunate restriction which will hopefully be lifted in the future.
+
+# Example fab commands
+
+Once you have a cluster you can use many different variations of the "fab" command to
+install Citus. Here are some examples:
+
+- `fab --list` will return a list of the tasks you can run.
+- `fab setup.basic_testing`, will create a vanilla cluster with postgres and citus. Once this has run you can simply run `psql` to connect to it.
+- `fab use.citus:v6.0.1 setup.basic_testing` will do the same, but use the tag `v6.0.1` when installing Citus. You can give it any git ref, it defaults to `master`.
+- `fab use.postgres:9.6.1 setup.basic_testing` lets you choose your postgres version.
+- `fab use.citus:v6.0.1 setup.enterprise` will install postgres and the `v6.0.1` tag of the enterprise repo.
+
+# Ordering of fab tasks
+
+Order is important though! There are three kinds of tasks, `use`, `setup`, and `add`. The
+`use` tasks don't take any actions, they just change some configuration which later tasks
+use. So, running `fab use.citus:v6.0.1` will do nothing. The `setup` tasks, such as
+`setup.enterprise` actually build and install things.
+
+Running `fab setup.enterprise use.citus:v6.0.1` is also a bad idea. The tasks are executed
+in order so this will first setup the `enteprise-master` branch of enterprise (because
+that's the default) then get ready to use the `v6.0.1` ref then exit.
+
+Finally, there are `add` tasks, such as `add.tpch`. These assume that a cluster is already
+installe and running (and therefore must come after the `setup` tasks, and install
+additional components, such as the tpch data.
+
+# `add` tasks
+
+It is possible to add extra extensions and features to a Citus cluster:
+
+- `fab add.tpch:scale_factor=1` will generate and stage tpch tables (the default scale factor is 10)
+- `fab add.session_analytics` will build and install the session_analytics package (it uses your ssh agent forwarding when connecting to github, so you must have the key you use for github added to your local agent)
+
+There's no need to run these as a separate command.
+
+- `fab use.citus:v6.0.1 setup.enterprise add.shard_rebalancer` does what you'd expect.
+
+# Advanced fab usage
+
+By default your fab commands configure the entire cluster, however they can be targeted
+at roles or individual machines.
+
+- `fab -R master pg.restart` will restart postgres on the master node
+- `fab -R workers pg.stop` will shutdown pg on all the workers.
```

Comment:
Yeah, that's a great idea, we can easily ask cloudformation to add these to `/etc/hosts`.